compiler/plugins/target/AMD-AIE/aie/AMDAIECreatePathFindFlows.cpp

```diff
@@ -287,16 +399,35 @@ LogicalResult runOnPacketFlow(
     }
   }
 
-  auto maybeRoutingConfiguration =
-      emitPacketRoutingConfiguration(deviceModel, packetFlows);
+  auto maybeRoutingConfiguration = emitPacketRoutingConfiguration(
+      deviceModel, packetFlows, priorPacketFlows);
   if (failed(maybeRoutingConfiguration)) {
     return device.emitOpError()
            << "could not create a valid routing configuration";
   }
   auto [masterSets, slaveAMSels] = maybeRoutingConfiguration.value();
-
   auto [slaveGroups, slaveMasks] = emitSlaveGroupsAndMasksRoutingConfig(
-      slavePorts, packetFlows, deviceModel.getPacketIdMaskWidth());
+      slavePorts, packetFlows, priorSlavePorts, priorPacketFlows,
+      deviceModel.getPacketIdMaskWidth());
+
+  // Erase any duplicate packet rules in exising, previously-routed packet
+  // flows.
+  std::vector<PacketRulesOp> priorPacketRules;
```

Comment:
I think `SmallVector` is preferred in general, not saying to start changing it everywhere, but just when you introduce new variables.